distributed/client.py

```diff
@@ -394,6 +396,16 @@ def __init__(self, address=None, start=True, loop=None, timeout=5,
         self.connection_args = self.security.get_connection_args('client')
         self._connecting_to_scheduler = False
 
+        if loop is None:
+            self._should_close_loop = None
+            if in_ioloop():
+                self.loop = IOLoop.current()
```

Comment:
My reasoning for it (which I'm happy to forget if it doesn't make sense) is that we often want to specify an IOLoop living on a different thread.  For example when we make a `LocalCluster` we also make a thread running an IOLoop.  We want to tell the client to use the same particular IOLoop running on that particular thread.